src/pj_fwd.c

```diff
@@ -1,58 +1,174 @@
-/* general forward projection */
-#define PJ_LIB__
-#include <proj.h>
-#include <projects.h>
-# define EPS 1.0e-12
-	XY /* forward projection entry */
-pj_fwd(LP lp, PJ *P) {
-	XY xy;
-    XY err;
-    double t;
-    int last_errno;
-
-    /* cannot const-initialize this due to MSVC's broken (non const) HUGE_VAL */
-    err.x = err.y = HUGE_VAL;
-
-    if (0==P->fwd)
-        return err;
-    last_errno = proj_errno_reset (P);
+/******************************************************************************
+ * Project:  PROJ.4
+ * Purpose:  Forward operation invocation
+ * Author:   Thomas Knudsen,  thokn@sdfe.dk,  2018-01-02
+ *           Based on material from Gerald Evenden (original pj_fwd)
+ *           and Piyush Agram (original pj_fwd3d)
+ *
+ ******************************************************************************
+ * Copyright (c) 2000, Frank Warmerdam
+ * Copyright (c) 2018, Thomas Knudsen / SDFE
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *****************************************************************************/
+#include <errno.h>
+
+#include "proj_internal.h"
+#include "projects.h"
+
+PJ_COORD pj_fwd_prepare (PJ *P, PJ_COORD coo) {
+    if (HUGE_VAL==coo.v[0])
+        return proj_coord_error ();
 
     /* Check validity of angular input coordinates */
     if (P->left==PJ_IO_UNITS_RADIANS) {
+        double t;
+        LP lp = coo.lp;
 
-        /* check for forward and latitude or longitude overange */
-        t = fabs(lp.phi)-M_HALFPI;
-        if (t > EPS || fabs(lp.lam) > 10.) {
-            pj_ctx_set_errno( P->ctx, -14);
-            return err;
+        /* check for latitude or longitude over-range */
+        t = (lp.phi < 0? -lp.phi: lp.phi) - M_HALFPI;
```

Comment:
I definitely disagree, but not so strongly as to refusing to introduce your suggestion (especially not since I had another operator formatting change to introduce in the same file)